unity-renderer/Assets/DCLFeatures/CameraReel/Gallery/Scripts/ThumbnailContextMenuController.cs

```diff
@@ -0,0 +1,97 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Browser;
+using DCL.Tasks;
+using DCLFeatures.CameraReel.Section;
+using DCLServices.CameraReelService;
+using DCLServices.EnvironmentProvider;
+using System;
+using System.Threading;
+using UnityEngine;
+
+namespace DCLFeatures.CameraReel.Gallery
+{
+    public class ThumbnailContextMenuController
+    {
+        private const string DELETE_ERROR_MESSAGE = "There was an unexpected error when deleting the picture. Try again later.";
+
+        private readonly ThumbnailContextMenuView view;
+        private CameraReelResponse picture;
+        private CancellationTokenSource deleteScreenshotCancellationToken;
+
+        public ThumbnailContextMenuController(ThumbnailContextMenuView view,
+            IClipboard clipboard,
+            CameraReelModel cameraReelModel,
+            IBrowserBridge browser,
+            ICameraReelStorageService storageService,
+            DataStore dataStore,
+            ICameraReelAnalyticsService analytics,
+            IEnvironmentProviderService environmentProviderService)
+        {
+            this.view = view;
+
+            view.OnSetup += p => picture = p;
+
+            view.OnCopyPictureLinkRequested += () =>
+            {
+                var url = $"https://reels.decentraland.{(environmentProviderService.IsProd() ? "org" : "zone")}/{picture.id}";
+                clipboard.WriteText(url);
+                analytics.Share("Explorer", "Copy");
+            };
+
+            view.OnDownloadRequested += () =>
+            {
+                browser.OpenUrl(picture.url);
+                analytics.DownloadPhoto("Explorer");
+            };
+
+            view.OnShareToTwitterRequested += () =>
+            {
+                var description = "Check out what I'm doing in Decentraland right now and join me!";
+                var url = $"https://reels.decentraland.{(environmentProviderService.IsProd() ? "org" : "zone")}/{picture.id}";
+                var twitterUrl = $"https://twitter.com/intent/tweet?text={description}&hashtags=DCLCamera&url={url}";
+
+                clipboard.WriteText(twitterUrl);
+                browser.OpenUrl(twitterUrl);
+                analytics.Share("Explorer", "Twitter");
+            };
+
+            view.OnDeletePictureRequested += () =>
```

Comment:
Personally I'd rather have the event subscription point to a named function to then unsubscribe in the dispose and have clearer code to read